src/ring_theory/noetherian.lean

```diff
@@ -360,6 +360,55 @@ begin
     by dsimp [gt]; simp only [lt_iff_le_not_le, (this _ _).symm]; tauto⟩
 end
 
+--in order to use the apply_instance, this line is needed
+open_locale classical
+/-- A module is Noetherian iff every nonempty set of submodules has a maximal submodule among them
+-/
+theorem set_has_maximal_iff_noetherian {R M} [ring R] [add_comm_group M] [module R M] : (∀(a : set $ submodule R M), a.nonempty → ∃ (M' ∈ a), ∀ (I ∈ a), M' ≤ I → I=M') ↔ is_noetherian R M :=
```

Comment:
Can this be split into a statement about well-foundedness in partial orders, chained with `is_noetherian_iff_well_founded`?

Reply:
To clarify, you want something of the form `well_founded r` implies the first definition [given here](https://en.wikipedia.org/wiki/Well-founded_relation)?